### DISCUSSION TITLE
if user is on unsupported chain after connecting wallet it should prompt user to switch to supported chain or disconnect wallet

### DISCUSSION THREAD

**Discussion by ethdev279** [WalletConnect/walletconnect-monorepo#4227] [Ideas]
if the user is on an unsupported chain after connecting the wallet(probably after page refresh, or opening the site that is already connected to wallet but user switched to unsupported chain) it should prompt the user to switch to a supported chain or disconnect the wallet. (not sure if it is already implemented)

just like how connectkit handles. 

![Screenshot 2024-02-13 at 10 08 10 PM](https://github.com/WalletConnect/walletconnect-monorepo/assets/45661693/0b6011f4-b3bc-48a6-a3c2-bfd3012de7bc)

**Comment by glitch-txs** [comment]
implemented in v4 :)
